File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/controller/NoticeController.java
package com.windea.demo.cloudcollect.core.controller;

import com.windea.demo.cloudcollect.core.domain.entity.Notice;
import com.windea.demo.cloudcollect.core.service.NoticeService;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.bind.annotation.*;

/**
 * 通知的控制器。
 */
@RestController
@RequestMapping("/notice")
@CrossOrigin
public class NoticeController {
	private final NoticeService service;

	public NoticeController(NoticeService service) {
		this.service = service;
	}


	//不允许用户自行创建通知

	@DeleteMapping("/{id}")
	public void delete(@PathVariable Long id) {
		service.delete(id);
	}

	@PutMapping("/{id}/read")
	public void read(@PathVariable Long id) {
		service.read(id);
	}

	@GetMapping("/{id}")
	public Notice get(@PathVariable Long id) {
		return service.get(id);
	}

	@GetMapping("/findAll")
	public Page<Notice> findAll(@RequestParam Pageable pageable) {
		return service.findAll(pageable);
	}

	@GetMapping("/findByUser")
	public Page<Notice> findByUser(@RequestParam Long userId, @RequestParam Pageable pageable) {
		return service.findByUser(userId, pageable);
	}

	@GetMapping("/findByUserAndReadStatus")
	public Page<Notice> findByUserAndReadStatus(@RequestParam Long userId, @RequestParam Boolean readStatus,
		@RequestParam Pageable pageable) {
		return service.findByUserAndReadStatus(userId, readStatus, pageable);
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/controller/NoticeController.kt
@file:Suppress("UNUSED_PARAMETER")

package com.windea.demo.cloudcollect.core.controller

import com.windea.demo.cloudcollect.core.domain.entity.*
import com.windea.demo.cloudcollect.core.service.*
import io.swagger.annotations.*
import org.springframework.data.domain.*
import org.springframework.security.access.prepost.*
import org.springframework.web.bind.annotation.*

@Api("通知")
@RestController
@RequestMapping("/notice")
@CrossOrigin
class NoticeController(
	private val noticeService: NoticeService
) {
	//@ApiOperation("创建一条通知。")
	//@PostMapping("/create")
	//@PreAuthorize("isAuthenticated()")
	//fun create(@RequestBody notice: Notice) {
	//	noticeService.create(notice)
	//}
	
	@ApiOperation("删除一条通知。")
	@DeleteMapping("/{id}")
	@PreAuthorize("isAuthenticated()")
	fun deleteById(@PathVariable id: Long) {
		noticeService.deleteById(id)
	}
	
	@ApiOperation("删除某一用户的所有浏览记录。")
	@DeleteMapping("/deleteAllByUserId")
	@PreAuthorize("isAuthenticated()")
	fun deleteAllByUserId(@RequestParam userId: Long) {
		noticeService.deleteAllByUserId(userId)
	}
	
	@ApiOperation("得到某一用户的所有通知。")
	@GetMapping("/findAllByUserId")
	fun findAllByUserId(@RequestParam userId: Long, pageable: Pageable): Page<Notice> {
		return noticeService.findAllByUserId(userId, pageable)
	}
}
